solr/core/src/test/org/apache/solr/util/TimeOutTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.util;
+
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.apache.solr.SolrTestCaseJ4;
+import org.apache.solr.common.util.TimeSource;
+import org.junit.BeforeClass;
+
+public class TimeOutTest extends SolrTestCaseJ4 {
+
+  @BeforeClass
+  public static void setUpOnce() {
+    assumeWorkingMockito();
+  }
+
+  public void testHasTimedOut() {
+    TimeSource mockTimeSource = mock(TimeSource.class);
```

Comment:
This could go in a Before method

Reply:
Ooh, good point!  Thanks, @madrob !

Reply:
I would do Before, not BeforeClass for it, and also put the when/thenReturn line in there as well.

Then in the tests adjust the timeouts to test various combinations if we have the same mock time source each time. Does that make sense?

Reply:
Okay!  I think I get it now.  Thanks for working with me on this @madrob !